Editorial review (#80)
iov-one/docs: docs/weave-tutorial/01-overview.md

> ---

Reviewer: Testing

> id: overview

Reviewer: ```suggestion
id: overview
```
```suggestion
id: overview test from jim
```

title: Tutorial Overview
sidebar_label: Overview
---

> First of all, to understand weave design philosophy and for the sake of programming well-designed software, I advise you to read this fine-grained article:

Reviewer 2: "First of all, to understand weave design philosophy and for the sake of programming well-designed software, I advise you to read this fine-grained article:"

"First of all, to understand the design philosophy behind Weave and in the interest of good software design, a recommended read is:"

> [Things I Learnt The Hard Way (in 30 Years of Software Development)](https://blog.juliobiason.net/thoughts/things-i-learnt-the-hard-way/)

Reviewer: `Cool article! I'm going to steal that for the documentation parts`

> To make the theory more tangible, we will build a sample application alongside this tutorial, to demonstrate dealing with real-world constraints. The application is located in the [tutorial](https://github.com/iov-one/tutorial/ "Tutorial Repository") repository, to show how to create a self-contained app.

Reviewer: ```suggestion
To make the theory more tangible, alongside this tutorial we will build a sample application, to demonstrate dealing with real-world constraints. The application, which is located in the [tutorial](https://github.com/iov-one/tutorial/ "Tutorial Repository") repository, shows how to create a self-contained app.
```

> In this tutorial, you will learn how to serialize and model your data structures, define messages and handlers, expose queries, and read initial configuration from the genesis file. You will be able to build a new extension and tie it together with other extensions into a complete blockchain application.

Reviewer: ```suggestion
In this tutorial, you will learn how to serialize and model your data structures, define messages and handlers, expose queries, and read initial configuration from the genesis file. In the end, you will be able to build a new extension and tie it together with other extensions into a complete blockchain application.
```

> Tutorial content is richened with Pull Request discussions. If you follow the PR reviews you can have a clear understanding of how to design your module and implement a blockchain app using weave piece by piece. And you can see how the development flow works and what are the issues you should have on your mind while moving forward.

Reviewer: ```suggestion
Tutorial content is enriched by Pull Request (PR) discussions. If you follow the PR reviews you will develop a clear understanding of how to design your module and implement a blockchain app using weave, piece by piece. And you can see how the development flow works and what are the issues you should have on your mind while moving forward.
```

> Whenever you have questions in your mind about the internals check out [Weave]("https://github.com/iov-one/weave) source code. It is very well documented. Please feel free to add [issues](https://github.com/iov-one/weave/issues) if you think there is something under-documented or confusing.

Reviewer: `I'm not sure what you mean by internals`

Author: Yes that's the correct wording

> Whenever you have questions in your mind about the internals check out [Weave]("https://github.com/iov-one/weave) source code. It is very well documented. Please feel free to add [issues](https://github.com/iov-one/weave/issues) if you think there is something under-documented or confusing.

Reviewer: ```suggestion
If you have questions about the internal working of Weave, you should check out the [Weave]("https://github.com/iov-one/weave) source code, which is very well documented. Please feel free to add to the list of [issues](https://github.com/iov-one/weave/issues) if you think there is something insufficiently documented or that is confusing.
```
